migrations/helpers/dart_samples_update_helper.py

```diff
@@ -0,0 +1,392 @@
+            the three expected fields doesn't exist), as you know that sample won't be in the process of being
+            cherrypicked. Although I can't think of a non-rare set of circumstances in which this will manifest as a
+            problem, and the one I can think of isn't that much of a problem (picking a sample that has just been
+            updated so it shouldn't be picked).
+            """
+            update_filtered_positive_fields(
+                filtered_positive_identifier,
+                samples,
+                version,
+                update_timestamp,
+            )
+
+            print("Updated filtered positives")
+
+            # 4. add the UUID fields if not present
+            add_sample_uuid_field(samples)
+
+            samples_updated_with_source_plate_uuids(mongo_db, samples)
+
+            # 5. update samples in mongo updated in either of the above two steps (would expect the same set of samples
+            #       from both steps)
+            print("Updating Mongo...")
+            _ = update_mongo_fields(mongo_db, samples, version, update_timestamp)
+            print("Finished updating Mongo")
+
+        # 6. update the MLWH (should be an idempotent operation)
+        # convert mongo field values into MySQL format
+        for sample in samples:
+            mongo_docs_for_sql.append(map_mongo_doc_to_sql_columns(sample))
+
+        if number_docs_found > 0:
+            print(f"Updating MLWH database for {len(mongo_docs_for_sql)} sample documents")
+            # create connection to the MLWH database
+            with create_mysql_connection(config, False) as mlwh_conn:
+
+                # execute SQL query to update filtered positive and UUID fields
+                run_mysql_executemany_query(
+                    mlwh_conn, SQL_MLWH_MULTIPLE_FILTERED_POSITIVE_AND_UUID_UPDATE, mongo_docs_for_sql
+                )
+        else:
+            print("No documents found for this timestamp range, nothing to insert or update in MLWH")
+
+        # 7. add all the plates of the positive samples we've selected in step 1 above, to DART
+        update_dart_fields(config, samples)
+    except Exception as e:
+        print(e)
+
+
+def get_positive_samples(
+    samples_collection: Collection, start_datetime: datetime, end_datetime: datetime
+) -> List[Sample]:
+    match = {
+        "$match": {
+            # 1. First filter by the start and end dates
+            FIELD_CREATED_AT: {"$gte": start_datetime, "$lte": end_datetime},
+            FIELD_RESULT: {"$regex": "^positive", "$options": "i"},
+            FIELD_ROOT_SAMPLE_ID: {"$not": {"$regex": "^CBIQA_"}},
+        }
+    }
+    return list(samples_collection.aggregate([match]))
+
+
+def remove_cherrypicked_samples(samples: List[Sample], cherry_picked_ids: List[str]) -> List[Sample]:
+    return list(filter(lambda sample: sample[FIELD_ROOT_SAMPLE_ID] not in cherry_picked_ids, samples))
```

Comment:
I still think that to remove cherrypicked samples you need to check on a (root sample id, plate barcode) tuple rather than just plate barcode. As multiple samples can exist with the same root sample id, with the potential for one being cherrypicked, and one not